HPX master built from source, cmake fails to link main.cpp example in docs

## Expected Behavior

Getting started helloworld manually typed example should build and run.
from here: https://stellar-group.github.io/hpx/docs/sphinx/tags/1.2.0/html/quickstart.html#hello-world

## Actual Behavior

link error:
``` 
main.cpp:(.text+0x44): undefined reference to `hpx::iostreams::flush'
```
here is output:
```
[grostig@fx8150-grostig-com build]$ cmake -DCMAKE_PREFIX_PATH=~/libs ..
-- The CXX compiler identification is GNU 8.2.1
-- Check for working CXX compiler: /usr/lib64/ccache/c++
-- Check for working CXX compiler: /usr/lib64/ccache/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The C compiler identification is GNU 8.2.1
-- Check for working C compiler: /usr/lib64/ccache/cc
-- Check for working C compiler: /usr/lib64/ccache/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Using tcmalloc allocator.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/grostig/src/hpx/helloworld_cmake/build
[grostig@fx8150-grostig-com build]$ make all
Scanning dependencies of target my_hpx_program
[ 50%] Building CXX object CMakeFiles/my_hpx_program.dir/main.cpp.o
[100%] Linking CXX executable my_hpx_program
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `main':
main.cpp:(.text+0x3a): undefined reference to `hpx::cout'
/usr/bin/ld: main.cpp:(.text+0x44): undefined reference to `hpx::iostreams::flush'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `void hpx::lcos::packaged_action<hpx::iostreams::server::output_stream::write_sync_action, void, false>::do_apply<unsigned int, unsigned long, hpx::iostreams::detail::buffer&>(hpx::naming::address&&, hpx::naming::id_type const&, hpx::threads::thread_priority, unsigned int&&, unsigned long&&, hpx::iostreams::detail::buffer&)':
main.cpp:(.text._ZN3hpx4lcos15packaged_actionINS_9iostreams6server13output_stream17write_sync_actionEvLb0EE8do_applyIJjmRNS2_6detail6bufferEEEEvONS_6naming7addressERKNSB_7id_typeENS_7threads15thread_priorityEDpOT_[_ZN3hpx4lcos15packaged_actionINS_9iostreams6server13output_stream17write_sync_actionEvLb0EE8do_applyIJjmRNS2_6detail6bufferEEEEvONS_6naming7addressERKNSB_7id_typeENS_7threads15thread_priorityEDpOT_]+0xd2): undefined reference to `char const* hpx::actions::detail::get_action_name<hpx::iostreams::server::output_stream::write_sync_action>()'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `bool hpx::apply_p_cb<hpx::iostreams::server::output_stream::write_sync_action, hpx::actions::typed_continuation<void, hpx::util::unused_type>, hpx::util::detail::bound_front<void (*)(boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> >, boost::system::error_code const&, hpx::parcelset::parcel const&), boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> > >, unsigned int, unsigned long, hpx::iostreams::detail::buffer&>(hpx::actions::typed_continuation<void, hpx::util::unused_type>&&, hpx::naming::address&&, hpx::naming::id_type const&, hpx::threads::thread_priority, hpx::util::detail::bound_front<void (*)(boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> >, boost::system::error_code const&, hpx::parcelset::parcel const&), boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> > >&&, unsigned int&&, unsigned long&&, hpx::iostreams::detail::buffer&)':
main.cpp:(.text._ZN3hpx10apply_p_cbINS_9iostreams6server13output_stream17write_sync_actionENS_7actions18typed_continuationIvNS_4util11unused_typeEEENS7_6detail11bound_frontIPFvN5boost13intrusive_ptrINS_4lcos6detail12promise_dataIvEEEERKNSC_6system10error_codeERKNS_9parcelset6parcelEEJSI_EEEJjmRNS1_6detail6bufferEEEEbOT0_ONS_6naming7addressERKNSZ_7id_typeENS_7threads15thread_priorityEOT1_DpOT2_[_ZN3hpx10apply_p_cbINS_9iostreams6server13output_stream17write_sync_actionENS_7actions18typed_continuationIvNS_4util11unused_typeEEENS7_6detail11bound_frontIPFvN5boost13intrusive_ptrINS_4lcos6detail12promise_dataIvEEEERKNSC_6system10error_codeERKNS_9parcelset6parcelEEJSI_EEEJjmRNS1_6detail6bufferEEEEbOT0_ONS_6naming7addressERKNSZ_7id_typeENS_7threads15thread_priorityEOT1_DpOT2_]+0xac): undefined reference to `char const* hpx::actions::detail::get_action_name<hpx::iostreams::server::output_stream::write_sync_action>()'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `hpx::actions::basic_action_impl<void (hpx::iostreams::server::output_stream::*)(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&), void (hpx::iostreams::server::output_stream::*)(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&), &hpx::iostreams::server::output_stream::write_sync, hpx::iostreams::server::output_stream::write_sync_action>::get_action_name[abi:cxx11](unsigned long)':
main.cpp:(.text._ZN3hpx7actions17basic_action_implIMNS_9iostreams6server13output_streamEFvjmRKNS2_6detail6bufferEESA_XadL_ZNS4_10write_syncEjmS8_EENS4_17write_sync_actionEE15get_action_nameB5cxx11Em[_ZN3hpx7actions17basic_action_implIMNS_9iostreams6server13output_streamEFvjmRKNS2_6detail6bufferEESA_XadL_ZNS4_10write_syncEjmS8_EENS4_17write_sync_actionEE15get_action_nameB5cxx11Em]+0x58): undefined reference to `char const* hpx::actions::detail::get_action_name<hpx::iostreams::server::output_stream::write_sync_action>()'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `std::enable_if<hpx::traits::is_continuation<hpx::actions::typed_continuation<void, hpx::util::unused_type>, void>::value, bool>::type hpx::detail::apply_cb_impl<hpx::iostreams::server::output_stream::write_sync_action, hpx::actions::typed_continuation<void, hpx::util::unused_type>, hpx::util::detail::bound_front<void (*)(boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> >, boost::system::error_code const&, hpx::parcelset::parcel const&), boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> > >, unsigned int, unsigned long, hpx::iostreams::detail::buffer&>(hpx::actions::typed_continuation<void, hpx::util::unused_type>&&, hpx::naming::id_type const&, hpx::threads::thread_priority, hpx::util::detail::bound_front<void (*)(boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> >, boost::system::error_code const&, hpx::parcelset::parcel const&), boost::intrusive_ptr<hpx::lcos::detail::promise_data<void> > >&&, unsigned int&&, unsigned long&&, hpx::iostreams::detail::buffer&)':
main.cpp:(.text._ZN3hpx6detail13apply_cb_implINS_9iostreams6server13output_stream17write_sync_actionENS_7actions18typed_continuationIvNS_4util11unused_typeEEENS8_6detail11bound_frontIPFvN5boost13intrusive_ptrINS_4lcos6detail12promise_dataIvEEEERKNSD_6system10error_codeERKNS_9parcelset6parcelEEJSJ_EEEJjmRNS2_6detail6bufferEEEENSt9enable_ifIXsrNS_6traits15is_continuationIT0_vEE5valueEbE4typeEOS11_RKNS_6naming7id_typeENS_7threads15thread_priorityEOT1_DpOT2_[_ZN3hpx6detail13apply_cb_implINS_9iostreams6server13output_stream17write_sync_actionENS_7actions18typed_continuationIvNS_4util11unused_typeEEENS8_6detail11bound_frontIPFvN5boost13intrusive_ptrINS_4lcos6detail12promise_dataIvEEEERKNSD_6system10error_codeERKNS_9parcelset6parcelEEJSJ_EEEJjmRNS2_6detail6bufferEEEENSt9enable_ifIXsrNS_6traits15is_continuationIT0_vEE5valueEbE4typeEOS11_RKNS_6naming7id_typeENS_7threads15thread_priorityEOT1_DpOT2_]+0xac): undefined reference to `char const* hpx::actions::detail::get_action_name<hpx::iostreams::server::output_stream::write_sync_action>()'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `int hpx::components::get_component_type<hpx::iostreams::server::output_stream>()':
main.cpp:(.text._ZN3hpx10components18get_component_typeINS_9iostreams6server13output_streamEEEiv[_ZN3hpx10components18get_component_typeINS_9iostreams6server13output_streamEEEiv]+0x5): undefined reference to `hpx::traits::component_type_database<hpx::iostreams::server::output_stream, void>::get()'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `std::enable_if<!hpx::traits::is_future<void>::value, void>::type hpx::actions::basic_action_impl<void (hpx::iostreams::server::output_stream::*)(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&), void (hpx::iostreams::server::output_stream::*)(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&), &hpx::iostreams::server::output_stream::write_sync, hpx::iostreams::server::output_stream::write_sync_action>::invoke_helper<void, unsigned int, unsigned long, hpx::iostreams::detail::buffer&>(hpx::actions::detail::tag<void>, unsigned long, int, unsigned int&&, unsigned long&&, hpx::iostreams::detail::buffer&)':
main.cpp:(.text._ZN3hpx7actions17basic_action_implIMNS_9iostreams6server13output_streamEFvjmRKNS2_6detail6bufferEESA_XadL_ZNS4_10write_syncEjmS8_EENS4_17write_sync_actionEE13invoke_helperIvJjmRS6_EEENSt9enable_ifIXntsrNS_6traits9is_futureIT_EE5valueESI_E4typeENS0_6detail3tagISI_EEmiDpOT0_[_ZN3hpx7actions17basic_action_implIMNS_9iostreams6server13output_streamEFvjmRKNS2_6detail6bufferEESA_XadL_ZNS4_10write_syncEjmS8_EENS4_17write_sync_actionEE13invoke_helperIvJjmRS6_EEENSt9enable_ifIXntsrNS_6traits9is_futureIT_EE5valueESI_E4typeENS0_6detail3tagISI_EEmiDpOT0_]+0x6c): undefined reference to `hpx::iostreams::server::output_stream::write_sync(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&)'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `hpx::actions::transfer_continuation_action<hpx::iostreams::server::output_stream::write_sync_action>::transfer_continuation_action<hpx::threads::thread_priority&, unsigned int, unsigned long, hpx::iostreams::detail::buffer&>(hpx::actions::typed_continuation<void, hpx::util::unused_type>&&, hpx::threads::thread_priority&, unsigned int&&, unsigned long&&, hpx::iostreams::detail::buffer&)':
main.cpp:(.text._ZN3hpx7actions28transfer_continuation_actionINS_9iostreams6server13output_stream17write_sync_actionEEC2IJRNS_7threads15thread_priorityEjmRNS2_6detail6bufferEEEEONS0_18typed_continuationIvNS_4util11unused_typeEEEDpOT_[_ZN3hpx7actions28transfer_continuation_actionINS_9iostreams6server13output_stream17write_sync_actionEEC5IJRNS_7threads15thread_priorityEjmRNS2_6detail6bufferEEEEONS0_18typed_continuationIvNS_4util11unused_typeEEEDpOT_]+0x7a): undefined reference to `vtable for hpx::actions::transfer_continuation_action<hpx::iostreams::server::output_stream::write_sync_action>'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `hpx::actions::detail::register_action<hpx::iostreams::server::output_stream::write_sync_action>::register_action()':
main.cpp:(.text._ZN3hpx7actions6detail15register_actionINS_9iostreams6server13output_stream17write_sync_actionEEC2Ev[_ZN3hpx7actions6detail15register_actionINS_9iostreams6server13output_stream17write_sync_actionEEC5Ev]+0x22): undefined reference to `char const* hpx::actions::detail::get_action_name<hpx::iostreams::server::output_stream::write_sync_action>()'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `hpx::actions::transfer_continuation_action<hpx::iostreams::server::output_stream::write_sync_action>::transfer_continuation_action()':
main.cpp:(.text._ZN3hpx7actions28transfer_continuation_actionINS_9iostreams6server13output_stream17write_sync_actionEEC2Ev[_ZN3hpx7actions28transfer_continuation_actionINS_9iostreams6server13output_stream17write_sync_actionEEC5Ev]+0x1a): undefined reference to `vtable for hpx::actions::transfer_continuation_action<hpx::iostreams::server::output_stream::write_sync_action>'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `hpx::actions::transfer_action<hpx::iostreams::server::output_stream::write_sync_action>::transfer_action()':
main.cpp:(.text._ZN3hpx7actions15transfer_actionINS_9iostreams6server13output_stream17write_sync_actionEEC2Ev[_ZN3hpx7actions15transfer_actionINS_9iostreams6server13output_stream17write_sync_actionEEC5Ev]+0x19): undefined reference to `vtable for hpx::actions::transfer_action<hpx::iostreams::server::output_stream::write_sync_action>'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `std::enable_if<!hpx::traits::is_future<void>::value, void>::type hpx::actions::basic_action_impl<void (hpx::iostreams::server::output_stream::*)(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&), void (hpx::iostreams::server::output_stream::*)(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&), &hpx::iostreams::server::output_stream::write_sync, hpx::iostreams::server::output_stream::write_sync_action>::invoke_helper<void, unsigned int, unsigned long, hpx::iostreams::detail::buffer>(hpx::actions::detail::tag<void>, unsigned long, int, unsigned int&&, unsigned long&&, hpx::iostreams::detail::buffer&&)':
main.cpp:(.text._ZN3hpx7actions17basic_action_implIMNS_9iostreams6server13output_streamEFvjmRKNS2_6detail6bufferEESA_XadL_ZNS4_10write_syncEjmS8_EENS4_17write_sync_actionEE13invoke_helperIvJjmS6_EEENSt9enable_ifIXntsrNS_6traits9is_futureIT_EE5valueESH_E4typeENS0_6detail3tagISH_EEmiDpOT0_[_ZN3hpx7actions17basic_action_implIMNS_9iostreams6server13output_streamEFvjmRKNS2_6detail6bufferEESA_XadL_ZNS4_10write_syncEjmS8_EENS4_17write_sync_actionEE13invoke_helperIvJjmS6_EEENSt9enable_ifIXntsrNS_6traits9is_futureIT_EE5valueESH_E4typeENS0_6detail3tagISH_EEmiDpOT0_]+0x6c): undefined reference to `hpx::iostreams::server::output_stream::write_sync(unsigned int, unsigned long, hpx::iostreams::detail::buffer const&)'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `hpx::actions::transfer_base_action<hpx::iostreams::server::output_stream::write_sync_action>::get_action_name() const':
main.cpp:(.text._ZNK3hpx7actions20transfer_base_actionINS_9iostreams6server13output_stream17write_sync_actionEE15get_action_nameEv[_ZNK3hpx7actions20transfer_base_actionINS_9iostreams6server13output_stream17write_sync_actionEE15get_action_nameEv]+0xd): undefined reference to `char const* hpx::actions::detail::get_action_name<hpx::iostreams::server::output_stream::write_sync_action>()'
/usr/bin/ld: CMakeFiles/my_hpx_program.dir/main.cpp.o: in function `unsigned int hpx::actions::detail::get_action_id<hpx::iostreams::server::output_stream::write_sync_action>()':
main.cpp:(.text._ZN3hpx7actions6detail13get_action_idINS_9iostreams6server13output_stream17write_sync_actionEEEjv[_ZN3hpx7actions6detail13get_action_idINS_9iostreams6server13output_stream17write_sync_actionEEEjv]+0x31): undefined reference to `char const* hpx::actions::detail::get_action_name<hpx::iostreams::server::output_stream::write_sync_action>()'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/my_hpx_program.dir/build.make:102: my_hpx_program] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/my_hpx_program.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

## Steps to Reproduce the Problem

... Please be as specific as possible while describing how to reproduce your problem.

>$  make all

Have attached some logs, let me know if you need more info. Thank you.
[CMakeLists.txt](https://github.com/STEllAR-GROUP/hpx/files/2846993/CMakeLists.txt)
[CMakeOutput.log](https://github.com/STEllAR-GROUP/hpx/files/2846995/CMakeOutput.log)



## Specifications

  - HPX Version:  github master current today hash is: cb2414c2d33bedd4ce2e9d918d47e300f4ac089a
  - Platform (compiler, OS): fedora29

--- a/docs/sphinx/quickstart.rst
+++ b/docs/sphinx/quickstart.rst
@@ -148,8 +148,8 @@ gentle introduction to the distributed aspects of |hpx|.
    It is also possible to build e.g. all quickstart examples using ``make
    examples.quickstart``.
 
-Setting up an |hpx| project
-===========================
+Hello, World!
+=============
 
 The following ``CMakeLists.txt`` is a minimal example of what you need in order to
 build an executable using |cmake|_ and |hpx|:
@@ -159,7 +159,9 @@ build an executable using |cmake|_ and |hpx|:
    cmake_minimum_required(VERSION 3.3.2)
    project(my_hpx_project CXX)
    find_package(HPX REQUIRED)
-   add_hpx_executable(my_hpx_program SOURCES main.cpp)
+   add_hpx_executable(my_hpx_program
+       SOURCES main.cpp
+       COMPONENT_DEPENDENCIES iostreams)
 
 .. note::
 
@@ -167,36 +169,28 @@ build an executable using |cmake|_ and |hpx|:
    See the section on :ref:`using_hpx_cmake` for more details on how to use
    ``add_hpx_executable``.
 
+.. note::
+
+   ``COMPONENT_DEPENDENCIES iostreams`` is optional for a minimal project but
+   lets us use the |hpx| equivalent of ``std::cout``, i.e. the |hpx|
+   :ref:`iostreams` functionality in our application.
+
 Create a new project directory and a ``CMakeLists.txt`` with the contents above.
-Then, in your project directory run the following:
-
-.. code-block:: sh
-
-   touch main.cpp # Create an empty file for now
-   mkdir build && cd build
-   cmake -DCMAKE_PREFIX_PATH=/path/to/hpx/installation ..
-   make all
-   ./my_hpx_program
-
-Your program should now start but fail quickly because it doesn't contain a main
-function yet.
-
-Hello, World!
-=============
-
-Once you have your project set up change the line containing
-``add_hpx_executable`` in ``CMakeLists.txt`` to the following:
-
-.. code-block:: cmake
-
-   add_hpx_executable(my_hpx_program SOURCES main.cpp COMPONENT_DEPENDENCIES iostreams)
-
-and add the following to your ``main.cpp``:
+Also create a ``main.cpp`` with the contents below.
 
 .. literalinclude:: ../examples/quickstart/simplest_hello_world_1.cpp
    :language: c++
    :start-after: //[simplest_hello_world_1_getting_started
    :end-before: //]
+
+Then, in your project directory run the following:
+
+.. code-block:: sh
+
+   mkdir build && cd build
+   cmake -DCMAKE_PREFIX_PATH=/path/to/hpx/installation ..
+   make all
+   ./my_hpx_program
 
 The program looks almost like a regular C++ hello world with the exception of
 the two includes and ``hpx::cout``. When you include ``hpx_main.hpp`` some

